src/Model/ModelManagerInterface.php

```diff
@@ -97,135 +80,100 @@ public function createQuery($class, $alias = 'o');
 
     /**
-     * Get a list of the field names models of the specified class use to store
-     * the identifier.
+     * Get a list of the field names models of the specified fully qualified
+     * class name used to store the identifier.
      *
-     * @param string $class fully qualified class name
-     *
-     * @return array
+     * @return string[]
      */
-    public function getIdentifierFieldNames($class);
+    public function getIdentifierFieldNames(?string $class): array;
 
     /**
      * Get the identifiers for this model class as a string.
-     *
-     * @param object $model
-     *
-     * @return string a string representation of the identifiers for this
-     *                instance
      */
-    public function getNormalizedIdentifier($model);
+    public function getNormalizedIdentifier(object $model): string;
 
     /**
      * Get the identifiers as a string that is safe to use in a url.
      *
      * This is similar to getNormalizedIdentifier but guarantees an id that can
      * be used in a URL.
-     *
-     * @param object $model
-     *
-     * @return string string representation of the id that is safe to use in a url
      */
-    public function getUrlSafeIdentifier($model);
+    public function getUrlSafeIdentifier(object $model): string;
 
     /**
      * Create a new instance of the model of the specified class.
-     *
-     * @param string $class
-     *
-     * @return object
      */
-    public function getModelInstance($class);
+    public function getModelInstance(string $class): object;
 
     /**
-     * @param string $class
-     *
-     * @return array|\ArrayAccess
+     * @return Collection<array-key, object>
      */
-    public function getModelCollectionInstance($class);
+    public function getModelCollectionInstance(string $class): Collection;
 
     /**
      * Removes an element from the collection.
      *
-     * @param array  $collection
-     * @param object $element
+     * @param Collection<array-key, object> $collection
      */
-    public function collectionRemoveElement(&$collection, &$element);
+    public function collectionRemoveElement(Collection &$collection, object &$element): void;
 
     /**
      * Add an element from the collection.
      *
-     * @param array  $collection
-     * @param object $element
+     * @param Collection<array-key, object> $collection
      */
-    public function collectionAddElement(&$collection, &$element);
+    public function collectionAddElement(Collection &$collection, object &$element): void;
 
     /**
      * Check if the element exists in the collection.
      *
-     * @param array  $collection
-     * @param object $element
-     *
-     * @return bool
+     * @param Collection<array-key, object> $collection
      */
-    public function collectionHasElement(&$collection, &$element);
+    public function collectionHasElement(Collection &$collection, object &$element): bool;
```

Comment:
If these arguments are objects, is there any reason to pass them as references?

Reply:
Fixed